### PR TITLE
fix: Rework exceptions to raise specific error types

### PR DESCRIPTION
**Pull request by don1989**
## Why
- The user might want to catch specific error types

### How
- Split errors into Client, Server, Validation errors
- Parse and raise the appropriate error type
- Start splitting tests into /unit and /e2e
- Start using unittest (what we're used to in original-backend)

### How Has This Been Tested?
- Locally
- via demo

### CODE REVIEW

File: original_sdk/client.py [hunk 1]
```diff
@@ -41,11 +46,15 @@ def set_http_session(self, session: requests.Session) -> None:
...
         except ValueError:
-            raise OriginalAPIException(response.text, response.status_code)
-        if response.status_code >= 399:
-            raise OriginalAPIException(response.text, response.status_code)
+            raise ClientError(
+                message="Invalid JSON received",
+                status=response.status_code,
+                data=response.text,
+            )
+        if is_error_status_code(response.status_code):
```
Comment (mgom) [inline original_sdk/client.py:56]: we should not combine both?

Comment (don1989) [reply original_sdk/client.py:56]: The types are slightly different for instance

- requests.Response uses `status`
- aiohttp.ClientResponse uses `status_code` 

Plus we need to await the async responses

Comment (don1989) [reply original_sdk/client.py]: @mgom updated to reuse as much as possible, but the sync/async differences means we need to keep some parts separate (getting the response details + `await`)

File: original_sdk/tests/test_async_client_e2e.py [hunk 2]
```diff
@@ -127,10 +127,12 @@ async def test_edit_asset(self, async_client: OriginalAsyncClient):
         while is_transferable is False and retries < TEST_RETRY_COUNTER:
             response = await async_client.get_asset(asset_uid)
             is_transferable = response["data"]["is_transferable"]
-            time.sleep(15)
+            await asyncio.sleep(15)
```
Comment (don1989) [inline original_sdk/tests/test_async_client_e2e.py:130]: Note: this seems to be the way to sleep when dealing with async (fixed flakey tests)

**Review by avmturo — APPROVED** [review]
LGTM

**Pull request merged**